#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, str_t

gen = ParameterGenerator()

gen.add("k_ratio", double_t, 0, "Specifies the delay in transform (tf) data that is tolerable in seconds.", 0.5, 0, 100)
gen.add("k_path", double_t, 0, "The frequency in Hz for the map to be updated.", 10, 0, 100)
gen.add("beta", double_t, 0, "The frequency in Hz for the map to be publish display information.", 10, 0, 100)

#map params
gen.add("lambda", double_t, 0, "The width of the map in meters.",0.5, 0, 100)
gen.add("acc_vx", double_t, 0, "The height of the map in meters.",0.5, 0, 2)
gen.add("acc_vth", double_t, 0, "The resolution of the map in meters/cell.", 0.5, 0,3)
gen.add("forward_max_vel_lin", double_t, 0, "The x origin of the map in the global frame in meters.",0.5,0,2)
gen.add("backward_max_vel_lin", double_t, 0, "The x origin of the map in the global frame in meters.",0.5,0,2)
gen.add("max_vel_th", double_t, 0, "The y origin of the map in the global frame in meters.", 0.5, 0,3)

# robot footprint shape
gen.add("min_vel_lin", double_t, 0, " ", 0.1,0,3 )
#gen.add("min_vel_th", double_t, 0, " ",  0.0,0,3)


exit(gen.generate("lane_follower", "lane_following_planner", "LaneFollowingPlanner"))

